src/proto/tagpost.proto

```diff
@@ -6,18 +6,71 @@ option java_multiple_files = true;
 option java_package = "com.google.tagpost";
 option java_outer_classname = "TagPostProto";
 
+import "google/protobuf/timestamp.proto";
+
 service TagpostService {
+  // dummy rpc service
   rpc FetchMessage(FetchMessageRequest) returns (FetchMessageResponse) {}
+  // fetch all threads by primary tag
+  rpc FetchThreadsByTag(FetchThreadsByTagRequest) returns (FetchThreadsByTagResponse) {}
+  // fetch all comments under one thread
+  rpc FetchCommentsUnderThread(FetchCommentsUnderThreadRequest) returns (FetchCommentsUnderThreadResponse) {}
+  // fetch all comments by a set of tags
+  rpc FetchCommentsByTags(FetchCommentsByTagsRequest) returns (FetchCommentsByTagsResponse) {}
+}
+
+message Tag {
+  string tag_name = 1;
+}
+
+message TagpostThread {
+  int64 thread_id = 1;
+  string primary_tag = 2;
+}
+
+message Comment {
+  int64 comment_id = 1;
+  int64 thread_id = 2;
+  string comment_content = 3;
+  google.protobuf.Timestamp timestamp = 4;
+  Tag primary_tag = 5;
+  string username = 6;
+  repeated Tag extra_tags = 7;
 }
 
 message FetchMessageRequest {
   string notice_key = 1;
   string request_id = 2;
   string user = 3;
-  string project = 4; 
+  string project = 4;
 }
 
 message FetchMessageResponse {
   string message = 1;
   repeated string comments = 2;
 }
+
+message FetchThreadsByTagRequest{
```

Comment:
Keep one blank character between Request and bracket. Also apply for other messages.